Minor refactoring of GradBucket class.

wayi1 opened the pull request:
Summary:
1. Moved the inline implementations of GradBucket class to the header for
succinctness and readability. This coding style is also consistent with
reducer.h under the same directory.

2. Changed the constructor of GradBucket to pass the input by const
reference and hence avoided unnecessary explicit move semantics. Since
previously the declaration and definition are separated, passing the input
tensor vector by value looks quite bizarre.

Test Plan: buck test caffe2/torch/lib/c10d:ProcessGroupGlooTest

pritamdamania87 commented:
Looks like there is some duplication between this PR and https://github.com/pytorch/pytorch/pull/44339? Should we close out https://github.com/pytorch/pytorch/pull/44339?

Commit:
Move the inline implementations of GradBucket class to the header.

Summary:
Moved the inline implementations of GradBucket class to the header for
succinctness and readability. This coding style is also consistent with
reducer.h under the same directory.

Test Plan: buck test caffe2/torch/lib/c10d:ProcessGroupGlooTest

Differential Revision: D23569701

fbshipit-source-id: 2ad9a72dc4b58dc0529c6557bb3b51385e416612

Commit:
Pass the input tensor vector by const reference. (#44340)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/44340

Changed the constructor of GradBucket to pass the input by const
reference and hence avoided unnecessary explicit move semantics. Since
previously the declaration and definition are separated, passing the input
tensor vector by value looks quite bizarre.

Test Plan: buck test caffe2/torch/lib/c10d:ProcessGroupGlooTest

Differential Revision: D23569939

fbshipit-source-id: 5467e85d7a4517f74a9991db59e740127046489d
